crates/pgt_lexer/src/lexed.rs

```diff
@@ -0,0 +1,107 @@
+use pgt_diagnostics::{Diagnostic, MessageAndDescription};
+use pgt_text_size::TextRange;
+
+use crate::SyntaxKind;
+
+/// Internal error type used during lexing
+#[derive(Debug, Clone)]
+pub struct LexError {
+    pub msg: String,
+    pub token: u32,
+}
+
+/// A specialized diagnostic for lex errors.
+#[derive(Clone, Debug, Diagnostic, PartialEq)]
+#[diagnostic(category = "syntax", severity = Error)]
+pub struct LexDiagnostic {
+    /// The location where the error is occurred
+    #[location(span)]
+    pub span: TextRange,
+    #[message]
+    #[description]
+    pub message: MessageAndDescription,
+}
+
+/// Result of lexing a string, providing access to tokens and diagnostics
+pub struct Lexed<'a> {
+    pub(crate) text: &'a str,
+    pub(crate) kind: Vec<SyntaxKind>,
+    pub(crate) start: Vec<u32>,
+    pub(crate) error: Vec<LexError>,
+    pub(crate) line_ending_counts: Vec<usize>,
+}
+
+impl Lexed<'_> {
+    /// Returns the number of tokens
+    pub fn len(&self) -> usize {
+        self.kind.len()
+    }
+
+    /// Returns true if there are no tokens
+    pub fn is_empty(&self) -> bool {
+        self.len() == 0
+    }
+
+    /// Returns an iterator over token kinds
+    pub fn tokens(&self) -> impl Iterator<Item = SyntaxKind> + '_ {
+        (0..self.len()).map(move |i| self.kind(i))
+    }
+
+    /// Returns the kind of token at the given index
+    pub fn kind(&self, idx: usize) -> SyntaxKind {
+        assert!(idx < self.len());
```

Comment:
willst du hier noch ne message anfügen? 
ansonsten wär's vllt. besser, einfach den access in `53` panicen zu lassen, dann kriegt man zumindest eine index-out-of-bounds meldung, oder?

Reply:
finds mit der message besser, weil so immer klar ist dass `idx` das problem ist, und nicht der vector.